### ISSUE TITLE
Add swiftlint in the CI

### ISSUE THREAD

**Issue by curquiza** [meilisearch/meilisearch-swift#77]
We should add a linter check in the CI.
- [x] update the CI
- [ ] update the CONTRIBUTING.md if another linter that swiftlint is chosen

**Comment by Ethenyl** [comment]
Hi there,

I would like to work on this issue, is it still available?

Thanks in advance 😉

**Comment by curquiza** [comment]
Yes it is! I assign you 😁

**Comment by Ethenyl** [comment]
Great! I might be needing a few more information before moving on. What is the expected output from the linter?

Do you expect annotations on changed files? FYI, there is a hard limit of 10 annotations imposed by **GitHub** (see this [link](https://github.community/t/maximum-number-of-annotations-that-can-be-created-using-github-actions-logging-commands/16991/2) for more details). It would be possible to make it happen only on files that are impacted by the PR.

Or would a simple failure would be enough if there are were SwiftLint errors introduced (supposing warnings are fine)?

Thanks in advance.

**Comment by curquiza** [comment]
We already have a `swiftlint` configuration (maybe @ppamorim can help you if you need to understand some configuration choices). The goal is only to add it to the CI.

- A simple failure is ok. We will see later if it's not really convenient.
- Let's say the warnings are ok for the moment
- I've just seen there are some files to correct because I get changes on `master` by doing `swiftlint autocorrect`. Probably my fault in my previous PRs because I don't use the linter for the moment since it's not in the CI.
- I've just seen there are errors in the config file:
```bash
$ swiftlint lint
Loading configuration from '.swiftlint.yml'
Invalid configuration for 'weak_delegate'. Falling back to default.
Found a configuration for 'type_body_length' rule, but it is disabled on 'disabled_rules'.
configuration error: 'prohibited_nan_comparison' is not a valid rule identifier
configuration error: 'only_after_dot' is not a valid rule identifier
configuration error: 'void_function_in_ternary' is not a valid rule identifier
configuration error: 'return_value_from_void_function' is not a valid rule identifier
[...]
```

I hope those lines answer your questions. Feel free to ask any questions again!
Thanks in advance for your help! 😁

**Comment by Ethenyl** [comment]
Thanks for the information 😄

I've been able to make the CI run **SwifLint** in two different ways (simple run & custom action), I'll do one more test and I will also try to clean invalid rules.

As for linting locally, it will be difficult to make it happen while using Xcode in Package mode since there are no build phases or script running.

Best solution is to create a `.pbxproj` file, open it for daily usage and open the whole folder in Package mode for manually checking that everything works for SPM.

**Comment by Ethenyl** [comment]
@curquiza do you have a preference between a simple shell script or using an open-source GitHub action (from which the shell script has been sourced)?

It produces the same result in the end, the sole difference is that it adds a dependency to an "external" action.

If you prefer to go with a script, don't worry I will be adding a comment with a credit to the original author.

**Comment by curquiza** [comment]
I would rather the script. I want the linter check to be added in the GHA, in [this file](https://github.com/meilisearch/meilisearch-swift/blob/master/.github/workflows/swift-test.yml), like this:
```yml
    [...]
    - name: Run tests
      run: swift test -v
    - name: Run linter
      run: swiftlint ....
```

**Comment by Ethenyl** [comment]
@curquiza it was clear enough to start working on it 😉 I had already created a branch and integrated all that was needed. I just wanted to be sure that I wasn't going in the wrong direction with the actual output of the Action itself.

I've added the linter job before the test one, this way, if the linting spawn any critical error, no need to wait for the test to complete.

Anyway, I will soon open a PR for that 😄

**Comment by curquiza** [comment]
> I've added the linter job before the test one, this way, if the linting spawn any critical error, no need to wait for the test to complete.

I completely agree!

Thanks for your clarification and your work! 😁

**Comment by Ethenyl** [comment]
I created #82 to implement this.

**Comment by bidoubiwa** [comment]
Fixed by #77
